src/ast.js

```diff
@@ -253,26 +253,22 @@ AST.prototype.resolvePrecedence = function(result, parser) {
       }
     }
   } else if (
-    (result.kind === "cast" &&
-      result.what &&
-      !result.what.parenthesizedExpression) ||
-    (result.kind === "silent" &&
-      result.expr &&
-      !result.expr.parenthesizedExpression)
+    ["silent", "cast"].includes(result.kind) &&
```

Comment:
sorry, I'm a bit maniac, could you rewrite the condition to be more optimized as this function is runned over all nodes it may impact a bit :

`(result.kind == 'silent' || result.kind == 'cast') && ...`

NB : The property check costs less than a function call and a lookup because you have only 2 entries

Reply:
Fixed! Might be a good idea to add performance testing in the future :wink:

Reply:
it has something similar :

```
 PASS  test/perf.test.js
  ● Console

    console.log test/perf.test.js:42
          + Tokens speed  => 205.15K/sec
    console.log test/perf.test.js:56
          + Reading speed => 437.1K/sec
    console.log test/perf.test.js:101
          + Nodes speed  => 79.69K/sec
    console.log test/perf.test.js:115
          + Overall speed => 211.43K/sec
```
It's not kilobytes but thousands of tokens or nodes dependending on what is tested.

The problem is that depends on the CPU & RAM frequency (disk does not enter into account) so you can't really know automatically it you break/degrade something - any idea is welcome

Reply:
Nice! I have no experience with performance testing myself, but I saw that `prettier/prettier` is doing [something similar](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md#performance) based on [Benchmark.js](https://www.npmjs.com/package/benchmark).